channel/app.go

```diff
@@ -163,6 +172,33 @@ func (d OptAppDec) Decode(r io.Reader) (err error) {
 	return errors.WithMessage(err, "resolve app")
 }
 
+// OptAppAndDataEnc makes an optional pair of App definition and Data encodable.
+type OptAppAndDataEnc struct {
```

Comment:
Why differentiate between `OptAppAndDataEnc` and `OptAppAndDataDec`? Which aspect of `OptAppAndDataEnc` is optional?

Reply:
The pair of "data" and "app" together are optional.

This was an existing construct, which I only moved it from one package to another. If we need to modify this, I think it is beyond the scope of this PR and we should do in a separate one.

Reply:
Created an issue #281